added rdf documents, and implemented fixes for errata 998 and 999

maximelefrancois86 opened the pull request:
To be double-checked by other editors first.

maximelefrancois86 committed:
added rdf documents, and implemented fixes for errata 998 and 999

dr-shorthair reviewed (approved):
I've inspected the .ttl versions of the RDF and these changes look fine.

maximelefrancois86 committed:
Merge branch 'gh-pages' of https://github.com/w3c/sdw into rdfxml

maximelefrancois86 committed:
implemented errata in spec document, and added changes

maximelefrancois86 commented:
Propose the following text to append to the errata document

6. Unsatisfiable classes in SSNX alignment module
Added: {$date}
Type: ???
Refers to: 6.2 SSNX Alignment Module
Description: Unsatisfiability of some classes due to axioms (1) and (2) being too strict:

sosa:hasResult ≡ oldssn:hasValue (1)
sosa:hasResult ≡ oldssn:observationResult (2)

Axiom (1) is causing the following problem: oldssn:hasValue is a sub-property of dul:hasRegion, whose range in DUL has dul:Abstract as a super-class. Yet, a sosa:Sampling procedure have only sosa:Sample as a result, and sosa:Sample is a sub-class of ( dul:Event or dul:Object or dul:Quality )

Axiom (2) is causing a few problem that are non trivial, but basically it boils down to either:

- disjunction between dul:SocialObject and dul:PhysicalObject .
- disjunctions between dul:Event and dul:Object .

Correction:  Axioms (1) and (2) are relaxed to (1') and (2') 

oldssn:hasValue ⊑ sosa:hasResult (1')
oldssn:observationResult ⊑ sosa:hasResult (2')


7. Unsatisfiable classes in SSNX alignment module
Added: {$date}
Type: ???
Refers to: 6.1 Dolce-Ultralite Alignment Module and 6.2 SSNX Alignment Module
Description: Unsatisfiability of some classes, ex. Survival classes: 

- class ssn-system:SurvivalProperty is a sub-class of ssn:Property, and has for inverseof( ssn-system:hasSurvivalProperty ) only (and at least one) ssn-system:SurvivalRange, that is a sub-class of ssn:Property
- inverseof( ssn-system:hasSurvivalProperty ) is a sub-property of ssn:isPropertyOf
ssn:Property has for ssn:isPropertyOf only sosa:FeatureOfInterest

So some instance of ssn-system:SurvivalRange would be both a ssn:Property and a sosa:FeatureOfInterest, which is forbidden.

Correction: A feature of interest can also be a quality, so axioms (1) and (2) are relaxed to (1') and (2') 👍 

sosa:FeatureOfInterest rdfs:subClassOf [ owl:unionOf ( dul:Event dul:Object dul:InformationEntity ) ] (1)
oldssn:FeatureOfInterest owl:equivalentClass sosa:FeatureOfInterest (2)

sosa:FeatureOfInterest rdfs:subClassOf dul:Entity (1')
oldssn:FeatureOfInterest rdfs:subClassOf sosa:FeatureOfInterest (2')

tidoust commented on ssn/index.html:
```diff
@@ -7604,6 +7604,20 @@ <h3>Changes since W3C Candidate Recommendation 11 July 2017 <a href="https://www
           <li>Removed Features at Risk section</li>
           <li>Removed Candidate Recommendation Exit Criteria section</li>
         </ol>
+        <h3>Changes implementing <a href="https://www.w3.org/2017/10/vocab-ssn-errata.html">Errata #6</a> since W3C Recommendation 19 October 2017 <a href="https://www.w3.org/TR/2017/REC-vocab-ssn-20171019/">
```
It's a good idea to maintain the "changes" section. Instead of adding an `<h3>` entry for all changes we will be making, I would suggest to have a "Changes since W3C Recommendation 19 October 2017" and list changes that have been made to the document on top of relaxing axioms too. Typically, I propose:

```html
<h3>Changes since W3C Recommendation 19 October 2017 <a href="https://www.w3.org/TR/2017/REC-vocab-ssn-20171019/">(https://www.w3.org/TR/2017/REC-vocab-ssn-20171019/)</a></h3>
<ol>
  <li>Fixed remaining broken links to ontology graph and namespaces and "specifically" typo</li>
  <li>Fixed Seismographs example, replacing the final <code>;</code> with a <code>.</code></li>
  <li>Fixed targets of <code>sosa:hosts</code> and <code>ssn:hasSubSystem</code> in DHT Deployment example</li>
  <li>Fixed domain/range for <code>actsOnProperty</code> and <code>isActedOnBy</code> in actuation diagrams</li>
  <li>Replaced inexistant <code>sosa:actuationMadeBy</code> by <code>sosa:madeByActuator</code> in apartment 134 example</li>
  <li>Relaxed axioms in 6.2 SSNX Alignment Module:
    <ul>
      <li>oldssn:FeatureOfInterest is a sub-class of sosa:FeatureOfInterest (instead of equivalent)</li>
      <li>oldssn:hasValue is a sub-property of sosa:hasResult (instead of equivalent)</li>
      <li>oldssn:observationResult is a sub-property of sosa:hasResult (instead of equivalent)</li>
    </ul>
  </li>
  <li>Relaxed axioms in 6.1 Dolce-Ultralite Alignment Module:
    <ul>
      <li>sosa:FeatureOfInterest is a sub-class of dul:Entity</li>
    </ul>
  </li>
</ol>
```

tidoust reviewed (commented):
See inline comment about changes section.

Also, I added the [errata document](https://github.com/w3c/sdw/blob/gh-pages/ssn/errata.html) to the repo. Could you add a couple of entries there that explain the changes that you made to the Editor's Draft and that should also be made to the Recommendation? (note the date at the top of the errata document should be updated too)

maximelefrancois86 committed:
Merge branch 'gh-pages' of https://github.com/w3c/sdw into rdfxml

maximelefrancois86 committed:
better description of changes in spec and errata document

maximelefrancois86 committed:
hasResultTime -> resultTime in htaccess document

tidoust committed:
Improve errata description

- Updated the date of the errata document
- Flagged errata as "Editorial" instead of "Changes to non-normative section"
- Added a note to explain meaning of the `≡` and `⊑` symbols
- Updated title of entry 7 to mention DUL

tidoust commented:
Thanks @maximelefrancois86, I slightly updated the errata descriptions, see last commit.

@arminhaller, I let you do a final review and merge this pull request. I'll update the published versions of the ontology files once done.

tidoust committed:
Fix "inexistant" typo

tidoust commented:
Now merging this pull request following email discussion with @arminhaller, who raised the typo fixed in latest commit and approved the pull request.
